### PR TITLE
Add email to firmware manager fail case

### PR DESCRIPTION
**Pull request by mwodahl**
# PR Details
Adds an email notification to the support team in the event of a firmware manager upgrade failure. 

## Description

This PR adds a check to make sure that the RSU is online before beginning the upgrade process. If at any point in the upgrade process the firmware manager fails an email notification will be sent to the support team with the RSU IP and the error. Additionally, if the post-upgrade script fails to execute on a Commsignia RSU an email will be sent as well with the RSU IP and error. 

## How Has This Been Tested?

This has been tested locally with a gmail account for both the firmware upgrade failure case and the Commsignia post-upgrade script failure case. 

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ X ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ X ] My changes require new environment variables:
  - [ X ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ X ] My changes require updates to the documentation:
  - [ X ] I have updated the documentation accordingly.
- [ ] My changes require updates and/or additions to the unit tests:
  - [ ] I have modified/added tests to cover my changes.
- [ X ] All existing tests pass.

### CODE REVIEW

File: services/addons/images/firmware_manager/upgrader.py [hunk 1]
```diff
@@ -71,6 +72,44 @@ def wait_until_online(self):
...
+        while iter < 5:
+            code = subprocess.run(
+                ["ping", "-n", "-c1", self.rsu_ip], capture_output=True
+            ).returncode
+            if code == 0:
+                return 0
+            iter += 1
+            time.sleep(1)
+        # 5 seconds pass with no response
+        return -1
```
Comment (dmccoystephenson) [inline services/addons/images/firmware_manager/upgrader.py]: When I initially saw this function being used, my first thought was that the 'check_online' function was returning the number of RSUs that were online. I see now that it returns 0 upon successful ping. I think the clarity of this code could be improved by returning True upon success and False upon failure.

**Review by dmccoystephenson — CHANGES_REQUESTED** [review]
I am seeing some test failures when running the unit tests with a virtual environment using Python 3.12.2.
image

I also left a comment regarding the clarity of a returned variable.

File: services/addons/images/firmware_manager/upgrader.py (hunk 1 above)
Comment (mwodahl) [reply services/addons/images/firmware_manager/upgrader.py]: Updated this to return true/false for clarity

**Comment by mwodahl** [comment]
> I am seeing some test failures when running the unit tests with a virtual environment using Python 3.12.2. image
> 
> I also left a comment regarding the clarity of a returned variable.

As for the testing failures I was unable to replicate these when running python 3.12.2 locally or when running python 3.12.2 in a virtual environment. I've attached a screenshot of the plugin versions I'm using when running these tests:
Screenshot 2024-05-10 at 8 56 59 AM

**Review by dmccoystephenson — APPROVED** [review]
This looks good to me & the unit tests pass!

**Review by drewjj — CHANGES_REQUESTED** [review]
This looks great. Although we are updating a lot of environment variables here. This basically will break our CDOT deployment the moment this gets merged if we don't update our helm scripts first. This repository also has example K8s yaml files. Please update the firmware manager one to include these.

**Comment by mwodahl** [comment]
> This looks great. Although we are updating a lot of environment variables here. This basically will break our CDOT deployment the moment this gets merged if we don't update our helm scripts first. This repository also has example K8s yaml files. Please update the firmware manager one to include these.

Updated the example firmware manager file to include the new environment variables and opened a devops PR to include the new environment variables in our deployments

**Review by drewjj — APPROVED** [review]
Looks good!

**Pull request merged**